### PR TITLE
Emit "FAILED: " in red if terminal supports ANSI color output

### PR DESCRIPTION
**Pull request by mqudsi**
Per [@jhasse's request](https://github.com/ninja-build/ninja/pull/1455#issuecomment-444128323)

### CODE REVIEW

**Comment by mqudsi** [comment]
Thanks, done.

**Comment by jhasse** [comment]
Thanks! Have you tested that this works on Windows? Should display a colored message on Windows 10 and the normal message on 7 and 8.

**Comment by mqudsi** [comment]
I tested it under Windows 10 and confirmed it worked there. I don't have a Windows 7/8 machine set up with a development toolchain, but I can probably just copy the binary and try it there directly.

**Comment by mqudsi** [comment]
@jhasse I've tested it on Windows 7 and can confirm that the latest build correctly avoids emitting the control codes under conhost without color support.

image

_Shown: without and with the latest commit to avoid color output if not supported_

File: src/build.cc [hunk 1]
```diff
@@ -138,7 +138,10 @@ void BuildStatus::BuildEdgeFinished(Edge* edge,
          o != edge->outputs_.end(); ++o)
       outputs += (*o)->path() + " ";
 
-    printer_.PrintOnNewLine("FAILED: " + outputs + "\n");
+    if (printer_.supports_color())
+        printer_.PrintOnNewLine("\x1B[31m" "FAILED: " "\x1B[0m" + outputs + "\n");
```
Comment (shahzadlone) [inline src/build.cc]: Perhaps a braced ```if``` might be better please.

File: src/build.cc [hunk 2]
```diff
@@ -138,7 +138,10 @@ void BuildStatus::BuildEdgeFinished(Edge* edge,
...
+    else
+        printer_.PrintOnNewLine("FAILED: " + outputs + "\n");
```
Comment (shahzadlone) [inline src/build.cc]: Perhaps a braced ```else``` might be better please.

File: src/build.cc (hunk 1 above)
Comment (mqudsi) [reply src/build.cc]: @shahzadlone My preference is for braced `if`/`else`, but I am sticking to the style of nearby code.

File: src/build.cc (hunk 2 above)
Comment (mqudsi) [reply src/build.cc]: @shahzadlone My preference is for braced `if`/`else`, but I am sticking to the style of nearby code.

**Pull request merged**